hie-docs-generator/src/Options.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Options where
+
+import qualified Data.Map as Map
+import           Haskell.Ide.ApplyRefactPlugin
+import           Haskell.Ide.Engine.BasePlugin
+import           Haskell.Ide.Engine.PluginDescriptor
+import           Haskell.Ide.ExamplePlugin2
+import           Haskell.Ide.ExamplePluginAsync
+import           Haskell.Ide.GhcModPlugin
+import           Haskell.Ide.HaRePlugin
+import           Options.Applicative
+
+data Config =
+  Config {plugins :: Plugins
+         ,prefix :: FilePath}
+
+config :: Parser Config
+config =
+  Config pluginList <$>
+  strOption (long "prefix" <> metavar "PREFIX" <>
+             help "directory where the docs are written to (defaults to current directory)")
+
+pluginList :: Plugins
```

Comment:
For documentation purposes I’ll repeat what I said in IRC regarding this issue:

Splitting up the plugins into a separate `hie-config` package results in a circular dependency since it needs to depend on `haskell-ide-engine` for the base plugin. On the other hand `haskell-ide-engine` (the executable, but that doesn’t seem to make a difference) needs to depend on `hie-config`. The base plugin can’t be split of because it depends on internals from `haskell-ide-engine` for the version command. Since we’ll rework the config for #25 anyway, it’s probably not worth putting too much work into this right now.